wallet/file-format.md

```diff
@@ -2,10 +2,115 @@
+
+> _NOTE:_ The nonce, used as initialization vector, must have a size of 96 bits (12 bytes) as recommended in Special Publication 800-38D.
+
+### Serialization
+
+An account, in a serialized form, consists of various components, including the account's private and public keys, a salt, a nonce, a version, and optionally, an address and nickname to aid in wallet recognition.
+The account should be serialized using YAML to enable human readability of the file's contents, particularly the address and nickname.
+
+The following table summarize the format:
+
+| Field | Presence | Format | Example |
+| ----- | -------- | ------- | ------- |
+| Version | Mandatory | Integer | 0 |
+| Nickname | Optional | String | "Savings" |
+| Address | Optional | String | "AU12..." |
+| PrivateKey | Mandatory | Byte array (ciphered) | [17, 42 ...] |
+| PublicKey | Mandatory | Byte array | [21, 126 ...] |
+| Salt | Mandatory | Byte array | [57, 125, 102, 235, 118, 62, 21, 145, 126, 197, 242, 54, 145, 50, 178, 98] |
+| Nonce | Mandatory | Byte array | [119, 196, 31, 33, 211, 243, 26, 58, 102, 180, 47, 57] |
+
+It is worth noting that the value of the Version field corresponds to the entire part of this specification version. Currently, the value should be set to 0.
+
+#### Example
+
+Here is an example of YAML serialization:
+
+```yaml
+---
+Version: 0
+Nickname: Savings
+Address: AU12...
+PrivateKey: [17, 42 ...]
+PublicKey: [21, 126 ...]
+Salt: [57, 125, 102, 235, 118, 62, 21, 145, 126, 197, 242, 54, 145, 50, 178, 98]
+Nonce: [119, 196, 31, 33, 211, 243, 26, 58, 102, 180, 47, 57]
+```
+
+## Security Concerns
+
+The wallet file format specification aims to ensure the security and integrity of user accounts. Then, several security concerns must be taken into account:
+
+### Encryption and Decryption
+
+The encryption and decryption of the private key must be performed correctly to ensure the confidentiality and integrity of the account. If the encryption algorithm or key derivation function is weak, the private key may be vulnerable to attacks. It is therefore crucial to use strong and proven encryption algorithms and key derivation functions.
+
+### Salt Generation
+
+The generation of random and unpredictable values for the salt is critical in protecting the private key. If the values are not truly random or can be easily guessed, the encryption of the private key may be compromised. It is therefore essential to use a robust and reliable method.
+
+### Password Strength
+
+The strength of the user's password is crucial to the security of the wallet. If the password is weak or easily guessable, the private key's confidentiality may be compromised. Therefore, it is essential to educate users on creating strong passwords and implementing password policies that enforce minimum complexity requirements.
```

Comment:
Altough i agree with this is general, I don't really understand the presence of this section, as there is no mention of password in this spec.

Reply:
see line 43 in PBKDF2 section.